.github/workflows/api.yml

```diff
@@ -10,16 +10,11 @@ jobs:
       - uses: actions/checkout@master
 
       - name: Install latest stable
-        uses: actions-rs/toolchain@v1
-        with:
-            toolchain: stable
-            override: true
+        uses: dtolnay/rust-toolchain@stable
```

Comment:
I'm not sure what is the best action at the moment. I looked at [nushell's workflow](https://github.com/nushell/nushell/blob/5d63f47c851ff23cdd78e9782372b23d1ab5ae89/.github/workflows/ci.yml#L44C15-L44C52) and they use this one [here](https://github.com/actions-rust-lang/setup-rust-toolchain) (which is also written in Rust).

Do you have any experience with both or one of the actions? Due to the [usage number of actions-rust-lang/setup-rust-toolchain](https://github.com/actions-rust-lang/setup-rust-toolchain/network/dependents) I'm in favor of using actions-rust-lang/setup-rust-toolchain.

Reply:
I have introduced `dtolnay/rust-toolchain` in several repositories related to rust-embedded, like
https://github.com/rust-embedded/embedded-hal/commit/ed9e0c13f134abed81225ec24f15d9cca996378f, about a year ago. I'm not aware of any issues, it "just works".

One interesting design decision with `dtolnay/rust-toolchain` is that one doesn't use a fixed version. Instead, the version specifier `@stable` references a tag (or branch?) that gets updated when new versions of `dtolnay/rust-toolchain` are released. That's of course questionable in terms of reproducibility. On the other hand, one can argue that this whole action is about interacting with external tooling that's not part of the current repository anyway, so having it automatically upgraded when necessary can make the build pipeline more stable.
And indeed, have a look at the history of nushell's workflow you linked to: `actions-rust-lang/setup-rust-toolchain` gets a new version every few months.

Without having used `actions-rust-lang/setup-rust-toolchain` I can't really compare both. I have no reason to believe that one or the other is better.

Reply:
I had another look into `actions-rust-lang/setup-rust-toolchain` and it says on the README:

> The action is heavily inspired by dtolnay's https://github.com/dtolnay/rust-toolchain and extends it with further features.

They also show the usage and there seems to be a moving target things:

> ```yaml
> - uses: actions-rust-lang/setup-rust-toolchain@v1
> ```

So it seems to me that using `actions-rust-lang/setup-rust-toolchain@v1` should do the trick.

Let's do it with `actions-rust-lang/setup-rust-toolchain@v1` and `on: [push, pull_request]`

BTW: with PREvant I try to follow the same strategy: 

> One interesting design decision with dtolnay/rust-toolchain is that one doesn't use a fixed version.

PREvant should stay backwards compatible all the time. Probably, that should be stated in the README. 🤔

Reply:
As mentioned in the commit message of https://github.com/aixigo/PREvant/pull/158/commits/7d3f9d543340307b13b85657e4aaed5fd2b34750, simply using actions-rust-lang/setup-rust-toolchain@v1 fails because it automatically sets `RUSTFLAGS = "-D warnings"` to deny all warnings.

@schrieveslaach do you want to keep that and fix all warnings before merging this, or should I manually set `rustflags` to an empty string?

https://github.com/actions-rust-lang/setup-rust-toolchain#rustflags

Reply:
There are some warnings in there due to deprecation attributes duet #23. So empty `RUSTFLAGS` would be best at the moment.

Reply:
Thanks a lot.